src/audio/dai-zephyr.c

```diff
@@ -569,6 +569,37 @@ static int dai_verify_params(struct dai_data *dd, struct comp_dev *dev,
 	return 0;
 }
 
+static int dai_get_dma_slot(struct dai_data *dd, struct comp_dev *dev, uint32_t *slot)
+{
+	struct dai_config cfg;
+	int ret;
+	int hs;
+
+	ret = dai_config_get(dd->dai->dev, &cfg, dev->direction);
+	if (ret < 0) {
+		comp_err(dev, "failed to fetch DAI configuration");
+		return ret;
+	}
+
+	hs = dai_get_handshake(dd->dai, dev->direction, dd->stream_id);
+	if (ret < 0) {
+		comp_err(dev, "failed to fetch DAI handshake");
+		return ret;
+	}
+
+	switch (cfg.type) {
+	case DAI_IMX_SAI:
+	case DAI_IMX_ESAI:
+		*slot = (hs & GENMASK(15, 8)) >> 8;
+		break;
+	default:
+		*slot = hs;
+		return 0;
```

Comment:
I'd either use `return 0;` in all `case`s or a `break` everywhere to keep it uniform

Reply:
fixed